src/backend/dwarf.c

```diff
@@ -1447,8 +1447,7 @@ void cv_outsym(symbol *s)
 #endif
             {
                 infobuf->writeByte(DW_OP_addr);
-                dwarf_addrel(infoseg,infobuf->size(),s->Sseg);
-                append_addr(infobuf, s->Soffset);    // address of global
+                dwarf_appreladdr(infoseg, infobuf, s->Sseg, s->Soffset); // address of global
```

Comment:
I guess this also cause the typical ld.gold issues. Do you have a bug report for this?